crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -1160,9 +1079,9 @@ mod tests {
             let error_result = response.unwrap_err();
             let error_object: ErrorObjectOwned = error_result.into();
 
-            let expected = expect!["-32000"];
+            let expected = expect!["-32001"];
             expected.assert_eq(&error_object.code().to_string());
-            let expected = expect!["Error executing mock db error"];
+            let expected = expect!["Unexpected error encountered while reading state."];
```

Comment:
this is less specific than the string it replaces - is there no known cause for the error? See above on Unexpected

Reply:
Yes, and this is intentional. The rpc caller will see, "Unexpected error...", and the error log will contain "error executing mock db error", or whatever the underlying error is

Reply:
As an example, encountering a StorageError means that either some TypedStoreError occurred w/ Rocksdb, or we are unable to deserialize the object from chain for whatever reason. In either case this info is not usable to the rpc caller, so we can just return a generic InternalError message